.travis.yml

```diff
@@ -27,7 +27,7 @@ jobs:
             env: CODING_STANDARDS=1
             install:
                 - mkdir -p "$HOME/.php-cs-fixer"
-                - "echo '{\"require\": {\"friendsofphp/php-cs-fixer\": \"^2.0\"}}' > \"$HOME/.php-cs-fixer/composer.json\""
+                - "echo '{\"require\": {\"friendsofphp/php-cs-fixer\": \"'$CS_FIXER_VERSION'\"}}' > \"$HOME/.php-cs-fixer/composer.json\""
```

Comment:
Woher kommt der wert dieser ENV var?

Reply:
Hier: https://travis-ci.org/redaxo/redaxo/settings
Idee ist, die Version anpassen zu können ohne Commit/PR.

Reply:
Brauchen wir das oder können wir einfacher auf eine neue version fest einstellen?


Die vars sind dort doch erstmal aehr versteckt

Reply:
Also alternativ können wir halt jetzt fest "2.7.1" reinsetzen, und dann erneut ein PR, wenn die Bugs im CS-Fixer gefixt sind, und wieder auf `^2.0` setzen. Also ohne Env-Vars.

An der Variante mit der Env-Var in den Travis-Settings finde ich smart, dass man sie ohne PR anpassen kann.
Bei der Variante mit der Version in der `.travis.yml` muss man halt immer erst einen PR machen, und die anderen PRs müssen rebasen, damit sich die Änderung auswirkt. Das fällt bei der Env-Var weg. Wenn also ein Problem mit einer neueren CS-F-Version auftritt, reicht es die Env-Var in den Settings anzupassen, und die Builds der PRs neuzustarten (ohne Rebase).

Reply:
Warum wird bei ^2.0 nicht die 2.7.1 installiert?

Hat sich unser problem mit release von 2.8. in den letzten tagen behoben?

Sowit ich weiß gab es rel. Selten probleme wg cs fixer, daher sehe ich den need für das umstellen noch nicht direkt

Reply:
Der Bug ist in 2.7.2, 2.7.3 und 2.8 drin (wurden alle in den letzten Tagen released).
2.7.1 ist also ein Downgrade auf die letzte funktionierende Version.

Ich hatte das Problem schon öfters, dass ich downgraden (oder stattdessen temporär manche Fixer deaktivieren) musste.

Was auch vorkommt: Es kommt eine neue Version raus, mit neuen Fixern innerhalb der von uns verwendeten Rulesets. Oder die vorhandenen Fixer wurden erweitert, sodass sie mehr Stellen auffinden. Auch dann sind unter Umständen alle Builds erstmal rot, und man könnte dann die Version temporär wieder festsetzen, bis man entschieden hat, ob man die neuen Fixer auch wirklich will (und dann einmal komplett drüber laufen lässt), oder ob man sie deaktiviert.

Aktuell geht es übrigens um diesen Bug: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3202

Reply:
Ok, dann mach wie du meinst